### DISCUSSION TITLE
enable data encryption in seaweedfs filer

### DISCUSSION THREAD

**Discussion by bijaypilabs** [seaweedfs/seaweedfs#5692] [Q&A] [answered]
How do i enable data encryption in seaweedfs filer, by updating the helm chart?

**Comment by bijaypilabs** [comment]
I tried by setting the encryptVolumeData parameter in values.yaml file but it does not work 
have taken the sample from here and set it to true: https://github.com/seaweedfs/seaweedfs/blob/610d397f76a372a1c5faa5fadfbf9eb8ff2dad1f/k8s/charts/seaweedfs/values.yaml#L497
@chrislusf

**Reply by chrislusf** [reply]
what did you see for `it does not work`?

**Reply by bijaypilabs** [reply]
so to test it if it works or not i tried these steps
1. upload a file using filer
2. get the metadata of the file to get its file id
3. get the file from volume server using the file id
since the file should have been stored in encrypted form, in the 3rd step i was able to get the file as it is from volume server which shows that it is not stored in encrypted format
if this process is wrong please help me with how can i test if the encryption happen or not? @chrislusf

**Reply by chrislusf** [reply]
what's the output of `fs.meta.cat /path/to/file` in `weed shell`?

**Reply by bijaypilabs** [reply]
am not able to connect to weed shell using command `weed shell -filer localhost:8888`
it returns
`master: localhost:9333 filer: localhost:8888
.....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................` on and on

**Reply by bijaypilabs** [reply]
even when i run it inside the cluster pod shell i get the same response @chrislusf

**Reply by chrislusf** [reply]
it shows `weed shell` could not connect to `localhost:9333`, or specifically the grpc port `localhost:19333`.

**Reply by bijaypilabs** [reply]
after port forwarding port localhost: 19333 i get
`....................I0624 14:56:21.611120 masterclient.go:228 master localhost:9333 redirected to leader seaweedfs-master-0.seaweedfs-master-headless.seaweedfs.svc.cluster.local:9333..............` continuosly @chrislusf

**Reply by chrislusf** [reply]
use `weed shell -master seaweedfs-master-0.seaweedfs-master-headless.seaweedfs.svc.cluster.local:9333`

**Reply by bijaypilabs** [reply]
still getting ................................... @chrislusf 
is there anything else i can try?

**Reply by bijaypilabs** [reply]
in logs am getting this error:
`I0624 16:50:40.747378 master_grpc_server.go:300 - client .adminShell@54bf3d04-72b1-454f-839b-fd6ff29357e4: rpc error: code = Canceled desc = context canceled
I0624 16:50:40.747513 master_grpc_server.go:379 - client .adminShell@54bf3d04-72b1-454f-839b-fd6ff29357e4`

**Comment by bijaypilabs** [comment, marked as answer]
was able to resolve above issue 
i was using values.yaml file from bitnami helm charts and they have not set up the encryptVolumeData field in their filer template so i have to clone it locally and update templates and then the encryption worked for me
to test if encryption work or not, i uploaded the file using filer and when i access the uploaded file using filer i get the unencrypted file but if i directly access the file using volume server i get encrypted file
to get the fileId for getting file from volume server you can get file metadata using this url format GET: http://127.0.0.1:8888/file?metadata=true
although the above weed shell related issue was not resolved for me we can have a separate discussing for it.

**Reply by chrislusf** [reply]
bitnami helm charts are different from SeaweedFS helm charts https://github.com/seaweedfs/seaweedfs/tree/master/k8s/charts/seaweedfs

What is the reason to pick the bitnami version?

**Reply by bijaypilabs** [reply]
the seaweedfs one was failing in my local due to some annotations that are set
